core/src/Streamly/Internal/FileSystem/PosixPath.hs

```diff
@@ -289,14 +319,14 @@ adapt p = fromPath (toPath p :: OS_PATH)
 -- "/"
 --
 {-# INLINE dropTrailingSeparators #-}
-dropTrailingSeparators :: OS_PATH -> OS_PATH
+dropTrailingSeparators :: OS_PATH_TYPE -> OS_PATH_TYPE
```

Comment:
The portable module could have used the definition directly, but it may not be possible to implement that in an easy way.